pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -192,6 +196,70 @@ type BMCDetails struct {
 	DisableCertificateVerification bool `json:"disableCertificateVerification,omitempty"`
 }
 
+// HardwareRAIDVolume defines the desired configuration of volume in hardware RAID
+type HardwareRAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	// +kubebuilder:validation:Minimum=0
+	SizeGibibytes *int `json:"sizeGibibytes,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0;1;1+0.
+	// +kubebuilder:validation:Enum="0";"1";"1+0"
+	Level nodes.RAIDLevel `json:"level" required:"true"`
```

Comment:
Since this type is just an alias for `string` and it seems to be the only reason to have added the dependency on the `notes` package from gophercloud, can we just use a string field here? That would let us avoid bringing in the gophercloud dependency, which is propagated to anything else that needs these types, like CAPM3 and HCC.

Reply:
Thanks, done.